src/BaseEntity.php

```diff
@@ -24,16 +26,32 @@
  * Since these objects aren't meant to be using any dependency, including logging, unit tests can and must be
  * written for each and all of their methods
  *
- * @property-read int $id
+ * @property-read int|null $id
  * @property-read int $uid
  * @property-read string $verb
- * @property-read int $type
+ * @property-read int|string $type
  * @property-read int $actorId
  * @property-read int $targetUriId
- * @property-read int $parentUriId
- * @property-read \DateTime $created
+ * @property-read int|null $parentUriId
+ * @property-read \DateTime|string $created
  * @property-read bool $seen
  * @property-read bool $dismissed
+ * @property-read string $name
+ * @property-read UriInterface $url
+ * @property-read UriInterface $photo
+ * @property-read DateTime $date
+ * @property-read string|null $msg
+ * @property-read UriInterface $link
+ * @property-read int|null $itemId
+ * @property-read int|null $parent
+ * @property-read string|null $otype
+ * @property-read string|null $name_cache
+ * @property-read string|null $msg_cache
+ * @property-read int|null $uriId
+ * @property-read string $cookie_hash
+ * @property-read string $user_agent
+ * @property-read bool $trusted
+ * @property-read string|null $last_used
```

Comment:
Shouldn't these added to the specific BaseEntity child class?

Reply:
PHPStan supports the `@property-read` annotation for defining read-only properties in classes. For extended classes, PHPStan will recognize `@property-read` annotations from the parent class. We can override or refine the type of a read-only property in a child class by using the `@property-read` annotation again in the child class.

Reply:
Yes, but this is an abstract class, which means there will never be an instance of it, so adding child-specific properties annotations in the parent abstract class seems counter-intuitive to me.

This also means that an IDE could suggest all of these properties for all the child classes instances, whether they make sense or not for the specific class.

Reply:
I was wrong. I can remove `@property-read` from the `BaseEntity.php`. The real problem was the missing property type in the child classes.

Reply:
Fixed.